From 1bd85e0413a950385efc8497115bc2c0bebdd67c Mon Sep 17 00:00:00 2001
From: Will Moss <willormos@gmail.com>
Date: Thu, 16 Mar 2023 17:45:46 +0530
Subject: [PATCH 073/281] OpenWrt:
 684-of_net-do-mac-address-increment-only-once.patch

of_net: do mac-address-increment only once

Remove mac-address-increment and mac-address-increment-byte
DT property after incrementing process to make sure MAC address
would not get incremented more if this function is stared again.
It could happen if device initialization is deferred after
unsuccessful attempt.

Signed-off-by: Will Moss <willormos@gmail.com>

Change-Id: Ibc58e9d97b1e8c2337f911b508ec2ce86347e1c7
Signed-off-by: Ram Chandra Jangir <quic_rjangir@quicinc.com>
---
 net/core/of_net.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/net/core/of_net.c b/net/core/of_net.c
index f1fa0aac725c..01ac1ca4f6ca 100644
--- a/net/core/of_net.c
+++ b/net/core/of_net.c
@@ -194,6 +194,12 @@ int of_get_mac_address(struct device_node *np, u8 *addr)
 		addr[3] = (mac_val >> 16) & 0xff;
 		addr[4] = (mac_val >> 8) & 0xff;
 		addr[5] = (mac_val >> 0) & 0xff;
+
+		/* Remove mac-address-increment and mac-address-increment-byte
+		 * DT property to make sure MAC address would not get incremented
+		 * more if this function is stared again. */
+		of_remove_property(np, of_find_property(np, "mac-address-increment", NULL));
+		of_remove_property(np, of_find_property(np, "mac-address-increment-byte", NULL));
 	}
 
 	of_add_mac_address(np, addr);
-- 
2.17.1

